chainerx_cc/chainerx/cuda/memory_pool_test.cc

```diff
@@ -116,20 +119,28 @@ class FixedCapacityDummyAllocator : public Allocator {
 
     MallocStatus Malloc(void** ptr, size_t bytesize) override {
         CHAINERX_ASSERT(bytesize > 0);
-        ++malloc_called_;
-        if (capacity_ < bytesize) {
-            return MallocStatus::kErrorMemoryAllocation;
+        uint8* mem{};
+        {
+            std::lock_guard<std::mutex> lock{sizes_mutex_};
+            ++malloc_called_;
+            if (capacity_ < bytesize) {
+                return MallocStatus::kErrorMemoryAllocation;
+            }
+            mem = new uint8_t[bytesize];
+            sizes_[mem] = bytesize;
+            capacity_ -= bytesize;
         }
-        // bytesize is encoded in the dummy pointer.
-        auto i = static_cast<intptr_t>(bytesize);
-        *ptr = reinterpret_cast<void*>(i);  // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
-        capacity_ -= bytesize;
+        *ptr = mem;
         return MallocStatus::kSuccess;
     }
     void Free(void* ptr) noexcept override {
-        intptr_t i = reinterpret_cast<intptr_t>(ptr);  // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
-        capacity_ += static_cast<size_t>(i);
-        ++free_called_;
+        uint8_t* mem = static_cast<uint8_t*>(ptr);
+        {
+            std::lock_guard<std::mutex> lock{sizes_mutex_};
+            capacity_ += sizes_.at(mem);
```

Comment:
How about deleting the entry from `sizes_` after `Free`? Not sure if we need to add an existence check but I think that's logically more correct (avoiding it from keep growing).

Reply:
Fixed.